Lib/fontmake/__main__.py

```diff
@@ -438,6 +440,31 @@ def main(args=None):
     glyphs_path = args.pop("glyphs_path")
     ufo_paths = args.pop("ufo_paths")
     designspace_path = args.pop("mm_designspace")
+    any_input = glyphs_path or ufo_paths or designspace_path
+
+    for file in unknown:
+        if any_input and not (ufo_paths and file.endswith(".ufo")):
+            logging.warn(
+                "Only one input file type accepted; additional file '%s' ignored" % file
+            )
+        elif file.endswith(".glyphs"):
+            glyphs_path = file
+            any_input = True
+        elif file.endswith(".designspace"):
+            designspace_path = file
+            any_input = True
+        elif file.endswith(".ufo"):
+            if not ufo_paths:
+                ufo_paths = []
+            ufo_paths.append(file)
+            any_input = True
+        else:
+            logging.warning("Unknown input file type '%s' ignored" % file)
+    if not any_input:
+        parser.print_help(sys.stderr)
+        print("fontmake: No input files specified")
+        sys.exit(1)
```

Comment:
you can use `parser.error()` here (see elsewhere in the file) instead of print + sys.exit